index.Rmd

```diff
@@ -429,6 +435,144 @@ You can also browse the history of a repository from your Git hosting service, a
 
 That's it: now you know how to keep track of all your files, so you will never lose them again, and no longer have to worry about making backups or saving multiple versions. In addition, this is the way that free and open-source code development happens in actuality. Also, the exercises and assignments in the course will be delivered and submitted this way, so make sure you are familiar with the whole process!
 
+## Project structure
+
+Another beneficial tool for scripting work is maintaining a consistent project structure as it will make it easier for you to switch from one project to the other and immediately understand how things work. In most cases, the project structure is entirely up to you. In some cases, however, the structure may be somewhat mandated, for instance, if you wish to make an R package. To make it easier to make a package out of your code, we will follow the structure of R packages:
+
+* A `main.R` script at the root of the project. This script performs step by step the different operations of your project. It is the only non-generic part of your project (it contains paths, already set variables, etc). If you are making a package, this should instead be a demo and live as `demo/main.R`.
```

Comment:
It would be good to make all these sections more language-neutral. They will first be learning Bash, so it's a bit odd to start with R, then Bash, then R again. However, we can include several examples. So here we can point out that R function files should go to `R`, Python function files should go to `Python` and Bash function files should go to `Bash`. Likewise, in addition to pointing out `source()` for R, we should include `import Python.filename` for Python.

Also, we can remove unnecessary complexity. The demo directory is not important.